src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -45,95 +42,158 @@ public final class ThrowableToErrorMapper {
-                      .findAny()
-                      .filter(
-                          t ->
-                              t instanceof AuthenticationException
-                                  || t instanceof IllegalArgumentException
-                                  || t instanceof NoNodeAvailableException)
-                      .orElse(null);
-              // connect to oss cassandra throws AuthenticationException for invalid credentials
-              // connect to AstraDB throws IllegalArgumentException for invalid token/credentials
-              if (error instanceof AuthenticationException
-                  || (error instanceof IllegalArgumentException
-                      && (error.getMessage().contains("AUTHENTICATION ERROR")
-                          || error
-                              .getMessage()
-                              .contains(
-                                  "Provided username token and/or password are incorrect")))) {
-                return ErrorCode.UNAUTHENTICATED_REQUEST
-                    .toApiException()
-                    .getCommandResultError(
-                        ErrorCode.UNAUTHENTICATED_REQUEST.getMessage(),
-                        Response.Status.UNAUTHORIZED);
-                // Driver NoNodeAvailableException -> ErrorCode.NO_NODE_AVAILABLE
-              } else if (error instanceof NoNodeAvailableException) {
-                return ErrorCode.NO_NODE_AVAILABLE
-                    .toApiException()
-                    .getCommandResultError(message, Response.Status.INTERNAL_SERVER_ERROR);
-              }
-            }
-          }
-          // Driver ClosedConnectionException -> ErrorCode.DRIVER_CLOSED_CONNECTION
-        } else if (throwable instanceof ClosedConnectionException) {
-          return ErrorCode.DRIVER_CLOSED_CONNECTION
-              .toApiException()
-              .getCommandResultError(message, Response.Status.INTERNAL_SERVER_ERROR);
-          // Unidentified Driver Exceptions, will not map into JsonApiException
-        } else if (throwable instanceof DriverException) {
-          if (throwable instanceof ReadFailureException) {
-            return ErrorCode.DATABASE_READ_FAILED
-                .toApiException(
-                    "root cause: (%s) %s", throwable.getClass().getName(), throwable.getMessage())
-                .getCommandResultError(message, Response.Status.INTERNAL_SERVER_ERROR);
-          }
-          return new CommandResult.Error(
-              message, fieldsForMetricsTag, fields, Response.Status.INTERNAL_SERVER_ERROR);
         }
 
+        // handle all driver exceptions
+        if (throwable instanceof DriverException) {
+          return handleDriverException(
+              (DriverException) throwable, message, fields, fieldsForMetricsTag);
+        }
+
+        // handle all other exceptions
         return new CommandResult.Error(message, fieldsForMetricsTag, fields, Response.Status.OK);
```

Comment:
Does this mean we are not returning a JsonApiException if we get to here? If it's an exception we don't recognize in any of our logic, should we really be returning HTTP OK?

Reply:
This return statement is our fallback in case there is an exception we do not recognize. We want to make a big deal out of this so that we can highlight the error for ourselves. Let's map it to `JsonApiException` with a new error code `SERVER_UNHANDLED_ERROR` and `Response.Status.INTERNAL_SERVER_ERROR` (aka 500).